File: 2000-2999/OpenJudge4067.java
import java.io.* ;
import java.util.* ;

public class Main {
	public static void main(String[] argv) {
		Scanner cin = new Scanner(System.in) ;
		while (cin.hasNext()) {
			String str = cin.nextLine() ;
			System.out.println(
				str.equals(new String(new StringBuilder(str).reverse()))?
				"YES":
				"NO") ;
		}
	}
}

File: 2000-2999/OpenJudge4067.cpp
#include <iostream>
#include <algorithm>
#include <string>  

using namespace std ;

int main(int argc, char *argv[])
{
	string str ;
	while (cin >> str) 
	{
		string str2 = str ;
		reverse(str.begin(), str.end());  
		puts(str == str2? "YES": "NO");
	}
	return 0 ;
}
